oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/IndexStructures.java

```diff
@@ -113,28 +113,39 @@ private Fields(Mappings mapping) {
          * Returns ture when the input fields have already been stored in the properties.
          */
         private boolean containsAllFields(Fields fields) {
-            return fields.properties.entrySet().stream()
-                                    .allMatch(item -> this.properties.containsKey(item.getKey()));
+            if (this.properties.size() < fields.properties.size()) {
+                return false;
+            }
+            boolean isContains = false;
+            isContains = fields.properties.entrySet().stream()
+                                    .allMatch(item -> {
+                                        Object hisItem = this.properties.get(item.getKey());
+                                        if (hisItem != null) {
+                                            return hisItem.toString().equals(item.getValue().toString());
+                                        } else {
+                                            return false;
+                                        }
+                                    });
+            if (!isContains) {
+                return false;
+            }
+
+            if (fields.source != null && this.source != null) {
+                isContains = this.source.getExcludes().toString().equals(fields.source.getExcludes().toString());
+            } else {
+                return fields.source == null && this.source == null;
+            }
+            return isContains;
```

Comment:
After you implement the `equals` method in `Source`, 

```suggestion
            return Objects.equals(this.source, fields.source);
```